tests/test_linalg.py

```diff
@@ -234,19 +234,408 @@ def test_llt_solve_shape_mismatch_errors(self):
+                "b must be state_size×0 when control_size = 0, but got "
+                "shape \\(2\\)"):
+            mm.KalmanFilterFp64(
+                x=x0, f=f, b=b_wrong, h=h, process_noise=0.1,
+                measurement_noise=1.0)
+
+    def test_control_2d_error(self):
+        x0 = mm.SimpleArrayFloat64([2])
+        f = mm.SimpleArrayFloat64([2, 2])
+        b_wrong = mm.SimpleArrayFloat64([1, 2])
+        h = mm.SimpleArrayFloat64([1, 2])
+
+        with self.assertRaisesRegex(
+                ValueError,
+                "KalmanFilterImpl::check_dimensions: The control SimpleArray "
+                "b must be state_size×control_size, but got shape \\(1, 2\\)"):
+            mm.KalmanFilterFp64(
+                x=x0, f=f, b=b_wrong, h=h, process_noise=0.1,
+                measurement_noise=1.0)
+
+
+class KalmanFilterPredictTC(unittest.TestCase):
+
+    def kf_predict_numpy(self, x, p, f, q):
+        x_pred = f @ x
+        p_pred = f @ p @ f.conj().T + q
+        return x_pred, p_pred
+
+    def test_predict_fp64(self):
+        n = 2
+        x0 = np.array([1.0, 2.0])
+        f = np.array([[1.1, 0.2],
+                      [0.1, 0.9]])
+        p0 = np.eye(n)
+        sigma_w = 0.316
+        q = (sigma_w**2) * np.eye(n)
+        x_pred_np, p_pred_np = self.kf_predict_numpy(x0, p0, f, q)
+
+        x_sa = mm.SimpleArrayFloat64(array=x0)
+        f_sa = mm.SimpleArrayFloat64(array=f)
+        h_sa = mm.SimpleArrayFloat64([1, n])
+        kf = mm.KalmanFilterFp64(
+            x=x_sa, f=f_sa, h=h_sa,
+            process_noise=sigma_w,
+            measurement_noise=1.0,
+        )
+        kf.predict()
+
+        x_pred_mm = kf.state.ndarray
+        p_pred_mm = kf.covariance.ndarray
+        np.testing.assert_allclose(x_pred_mm, x_pred_np, atol=1e-12, rtol=0.0)
+        np.testing.assert_allclose(p_pred_mm, p_pred_np, atol=1e-12, rtol=0.0)
+
+    def test_predict_cp128(self):
+        n = 2
+        x0 = np.array([1.0+0.5j, 2.0-0.3j], dtype=np.complex128)
+        f = np.array([[1.1+0.1j, 0.2-0.1j],
+                      [0.1+0.1j, 0.9+0.0j]], dtype=np.complex128)
+        p0 = np.eye(n, dtype=np.complex128)
+        sigma_w = 0.316
+        q = (sigma_w**2) * np.eye(n, dtype=np.complex128)
+        x_pred_np, p_pred_np = self.kf_predict_numpy(x0, p0, f, q)
+
+        x_sa = mm.SimpleArrayComplex128(array=x0)
+        f_sa = mm.SimpleArrayComplex128(array=f)
+        h_sa = mm.SimpleArrayComplex128([1, n])
+        kf = mm.KalmanFilterComplex128(
+            x=x_sa, f=f_sa, h=h_sa,
+            process_noise=sigma_w,
+            measurement_noise=1.0,
+        )
+        kf.predict()
+
+        x_pred_mm = kf.state.ndarray
+        p_pred_mm = kf.covariance.ndarray
+        np.testing.assert_allclose(x_pred_mm, x_pred_np, atol=1e-12, rtol=0.0)
+        np.testing.assert_allclose(p_pred_mm, p_pred_np, atol=1e-12, rtol=0.0)
+
+
+# The bug is reported in issue #604:
```

Comment:
Turn the two example functions `_wrap_complex()` and `sa_from_np()` as ordinary test functions (or classes) and decorate them as [skip or expected (known) failure](https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures).

Keep the comment with the new test code too.